README.md

```diff
@@ -153,76 +153,135 @@ On certain pages an API endpoint is called to receive experiment analysis data f
+Start all containers in the background (not attached to shell).  They can be stopped using `make kill`.
 
-### check
+### Running tests and checks
 
-Run all test and lint suites, this is run in CI on all PRs and deploys
+#### make check
 
-### py_test
+Run all test and lint suites, this is run in CI on all PRs and deploys.
 
-Run python tests
+#### make py_test
 
-### migrate
+Run only the python test suite.
 
-Apply all django migrations
+#### make bash
 
-### load_locales_countries
+Start a bash shell inside the container.  This lets you interact with the containerized filesystem and run Django management commands.
 
-Populates locales and countries
+##### Helpful Python Tips
+You can run the entire python test suite without coverage using the Django test runner:
 
-### load_dummy_experiments
+        ./manage.py test
 
-Populates db with dummy experiments
+For faster performance you can run all tests in parallel:
 
-### bash
+        ./manage.py test --parallel
 
-Start a bash shell inside the container (this lets you interact with the containerized filesystem and run Django management commands)
+You can run only the tests in a certain module by specifying its Python import path:
 
-### ssl
+        ./manage.py test experimenter.experiments.tests.api.v5.test_serializers
 
-Create dummy SSL certs to use the dev server over a locally secure
-connection. This helps test client behaviour with a secure
-connection. This task is run automatically when needed.
+For more details on running Django tests refer to the [Django test documentation](https://docs.djangoproject.com/en/3.1/topics/testing/overview/#running-tests)
 
-### kill
+To debug a test, you can use ipdb by placing this snippet anywhere in your code, such as within a test method or inside some application logic:
+
+        import ipdb
+        ipdb.set_trace()
+
+Then invoke the test using its full path:
+
+        ./manage.py experimenter.some_module.tests.some_test_file.SomeTestClass.test_some_thing
+
+And you will enter an interactive iPython shell at the point where you placed the ipdb snippet, allowing you to introspect variables and call methods
+
+For coverage you can use pytest, which will run all the python tests and track their coverage, but it is slower than using the Django test runner:
+
+        pytest --cov --cov-report term-missing
+
+You can also enter a Python shell to import and interact with code directly, for example:
+
+        ./manage.py shell
+
+And then you can import and execute arbitrary code:
+
+        from experimenter.experiments.models import NimbusExperiment
+        from experimenter.experiments.tests.factories import NimbusExperimentFactory
+        from experimenter.kinto.tasks import nimbus_push_experiment_to_kinto
+
+        experiment = NimbusExperimentFactory.create_with_status(NimbusExperiment.Status.DRAFT, name="Look at me, I'm Mr Experiment")
+        nimbus_push_experiment_to_kinto(experiment.id)
```

Comment:
On all these code blocks can we use triple backticks? Also allows us to specify syntax for highlighting. Currently they are all unnecessarily indented:

![Screen Shot 2021-03-31 at 2 17 05 PM](https://user-images.githubusercontent.com/6392049/113191627-c74ea080-922b-11eb-901c-2c8eaaf52738.png)

versus

![Screen Shot 2021-03-31 at 2 19 15 PM](https://user-images.githubusercontent.com/6392049/113191899-1c8ab200-922c-11eb-98aa-86dbea223452.png)

Reply:
+1, I left a similar comment further up before seeing Jody's here.